### ISSUE TITLE
derive_more::Add and another Add trait impl: spurious error

### ISSUE THREAD

**Issue by antifuchs** [intellij-rust/intellij-rust#4629]
## Environment

* **IntelliJ Rust plugin version:** 0.2.109.2149-192
* **Rust toolchain version:** 1.39.0
* **IDE name and version:** clion 2019.2
* **Operating system:** macOS

## Problem description
Using `derive_more`'s Add derive with another impl of `Add` for a different type, the rust plugin complains about implementations of `Add` that `derive_more` generates.

Cargo `test`, `build` and `check` on the source code that the plugin complains about does work, though.

## Steps to reproduce

This is probably best illustrated with this snippet of code:

``` rust
use derive_more::Add;
use std::ops::Add;
use std::time::Duration;

#[derive(PartialEq, Eq, Default, Debug, Clone, Copy, PartialOrd, Ord, Add)]
struct Nanos(u64);

// Commenting impl this out also makes the error go away:
impl Add<Duration> for Nanos {
    type Output = Nanos;

    fn add(self, rhs: Duration) -> Self::Output {
        let rhs: Nanos = Nanos(rhs.as_nanos() as u64);
        Nanos(rhs.0 + self.0)
    }
}

// Manually impl-ing works:
/*
impl Add<Nanos> for Nanos {
    type Output = Nanos;

    fn add(self, rhs: Nanos) -> Self::Output {
        Nanos(self.0 + rhs.0)
    }
}
*/

fn main() {
    let x = Nanos(4);
    let y = Nanos(1);
    println!("sum: {:?}", x + y); // <- spurious error on `y`
}
```

(this uses `derive_more = "0.99.1"` as a dependency in Cargo.toml.)

The rust plugin reports "mismatched types [E0308]: expected `Duration`, found `Nanos`" on the `x + y` line at the bottom. 

As mentioned in the code above, manually impl-ing the Add, or removing the Add impl for the other type makes the plugin-reported error go away.
